internal/pkg/api/server_test.go

```diff
@@ -349,7 +349,7 @@ key: %s`,
 		cfg.TLS = tlsCFG
 
 		st := NewStatusT(cfg, nil, nil)
-		srv := NewServer(addr, cfg, nil, nil, nil, nil, st, sm, fbuild.Info{}, nil, nil, nil, nil, nil)
+		srv := NewServer(addr, cfg, nil, nil, nil, nil, st, sm, fbuild.Info{}, nil, nil, nil, nil, nil, nil)
```

Comment:
This has become a ton of parameters will most being `nil`. Switching to a struct as a parameter would be better and cleaner. (don't do it in this PR, but an overall improvement)

Reply:
Created https://github.com/elastic/fleet-server/issues/3823 to track